ddtrace/context.py

```diff
@@ -125,9 +132,56 @@ def _traceparent(self):
         # type: () -> str
         if self.trace_id is None or self.span_id is None:
             return ""
+        tp = self._meta.get(_TRACEPARENT_KEY)
```

Comment:
Do we get the traceparent from the meta dictionary to avoid encoding the trace id multiple times? If so what's the expected performance improvement? This change make this function harder to read so I want to know if it's worth it.

Reply:
We need to pull out the original traceparent so that we can grab the original hexadecimal traceparent trace id. Since we can't convert back from dd trace id -> traceparent trace id, because the dd trace id only represents the lower 64 bits of the 128 digit hexadecimal traceparent id.

Reply:
Ahh that makes a lot of sense. I missed your comment bellow